### PR TITLE
[Lock] let BlockingStoreInterface extend PersistingStoreInterface

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/32555#issuecomment-515960229
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/Lock/Tests/Store/CombinedStoreTest.php [hunk 1]
```diff
@@ -62,8 +62,8 @@ public function getStore()
     protected function setUp()
     {
         $this->strategy = $this->getMockBuilder(StrategyInterface::class)->getMock();
-        $this->store1 = $this->getMockBuilder([PersistingStoreInterface::class, BlockingStoreInterface::class])->getMock();
-        $this->store2 = $this->getMockBuilder([PersistingStoreInterface::class, BlockingStoreInterface::class])->getMock();
+        $this->store1 = $this->getMockBuilder(BlockingStoreInterface::class)->getMock();
+        $this->store2 = $this->getMockBuilder(BlockingStoreInterface::class)->getMock();
```
Comment (OskarStark) [inline src/Symfony/Component/Lock/Tests/Store/CombinedStoreTest.php]: Is there a reason to not use “createMock()” ?

Comment (fabpot) [reply src/Symfony/Component/Lock/Tests/Store/CombinedStoreTest.php]: We should use `createMock()`. Existing tests might use this construct, but that's for historical reasons.

**Review by jderusse — COMMENTED** [review]
Shouldn't we remove `PersistingStoreInterface` from stores who implements both interfaces?

https://github.com/symfony/symfony/blob/a7852c0da89d926889c430d57e91f740dd6f1b62/src/Symfony/Component/Lock/Store/RetryTillSaveStore.php#L29

**Comment by xabbuh** [comment]
@jderusse fixed

**Review by jderusse — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**